src/core/utils/omit-deep.ts

```diff
@@ -1,9 +1,10 @@
 import type { GoslingSpec } from '@gosling.schema';
-import { cloneDeepWith, type CloneDeepWithCustomizer } from 'lodash';
+import { cloneDeepWith } from 'lodash-es';
+import { isObject } from '../gosling.schema.guards';
 
 export function omitDeep(spec: GoslingSpec, omitKeys: string[]) {
-    return cloneDeepWith(spec, (value: CloneDeepWithCustomizer<GoslingSpec>) => {
```

Comment:
The type of `CloneDeepWithCustomizer<GoslingSpec>` is as function (not an object), so previously the type was `never` in:

```typescript
        if (value && typeof value === 'object') {
            // value: never
            omitKeys.forEach(key => {
                delete value[key];
            });
```

So typescript never complained. This new typing narrows the type of value to `Record<PropertyKey, unknown>`, which is more likely what we wanted.